qmetaobject_impl/src/doc.rs

```diff
@@ -0,0 +1,100 @@
+/* Copyright (C) 2020 ivan tkachenko a.k.a. ratijas <me@ratijas.tk>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+associated documentation files (the "Software"), to deal in the Software without restriction,
+including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial
+portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+use darling::FromMeta;
+use proc_macro::TokenStream;
+use syn::{AttributeArgs};
+
+#[derive(Default, FromMeta)]
+#[darling(default)]
+pub struct QtDocMetaArgs {
+    qt: String,
+    // exactly one of them must be present at a time.
```

Comment:
I suppose this cannot be an enum because darling won't support it?

Reply:
Yeah(((
It would require something like `#[serde(untagged)]` in conjunction with `#[serde(flatten)]` in `darling` world.
It this this structure should be further parsed into a struct with enum, to enforce restriction on a type-level. But not sure how to report errors yet.